prow/cmd/plank/main.go

```diff
@@ -96,22 +98,28 @@ func main() {
 
 	pjutil.ServePProf()
 
-	configAgent := &config.Agent{}
+	var configAgent config.Agent
 	if err := configAgent.Start(o.configPath, o.jobConfigPath); err != nil {
 		logrus.WithError(err).Fatal("Error starting config agent.")
 	}
 	cfg := configAgent.Config
 
-	secretAgent := &secret.Agent{}
-	if o.github.TokenPath != "" {
-		if err := secretAgent.Start([]string{o.github.TokenPath}); err != nil {
-			logrus.WithError(err).Fatal("Error starting secrets agent.")
+	var reporter func(context.Context)
+	if !o.skipReport {
+		logrus.Warn("Plank no longer supports github reporting, migrate to crier before June 2020")
+		prowjobClientset, err := o.kubernetes.ProwJobClientset(cfg().ProwJobNamespace, o.dryRun)
+		if err != nil {
+			logrus.WithError(err).Fatal("Unable to create prowjob client")
 		}
-	}
 
-	githubClient, err := o.github.GitHubClient(secretAgent, o.dryRun)
-	if err != nil {
-		logrus.WithError(err).Fatal("Error getting GitHub client.")
+		const resync = 0 * time.Minute
+		prowjobInformerFactory := prowjobinformer.NewSharedInformerFactoryWithOptions(prowjobClientset, resync, prowjobinformer.WithNamespace(cfg().ProwJobNamespace))
+		controller, err := githubreporter.Controller(o.github, o.dryRun, cfg, "", prowjobClientset, prowjobInformerFactory.Prow().V1().ProwJobs(), 1)
+		if err != nil {
+			logrus.WithError(err).Fatal("Error creating github reporter")
+		}
+		reporter = controller.Run
```

Comment:
Why? We can just leave it as-is and remove all of it in July.

This change means plank ends up having two caches for Prowjobs if it has reporting enabled.

Reply:
Can't it share a cache? And it isn't super relevant as people shouldn't be doing this. 

This allows us to simplify the code now.

Reply:
It can not, unless you change crier to be a controller-runtime based controller (which would be a good idea anyways, as it would get rid of all the boilerplate needed for manually written controllers)

Reply:
Ack